src/api/app/helpers/webui/staging/workflow_helper.rb

```diff
@@ -23,10 +23,11 @@ def review_progress(staging_project)
   end
 
   def testing_progress(staging_project)
-    notdone = staging_project.checks.pending.size
     allchecks = staging_project.checks.size + staging_project.missing_checks.size
+    return 0 if allchecks == 0
```

Comment:
why are you returning 0 instead of 100 now?

Reply:
testing_progress shouldn't even be called in that situation - as the staging project would be in a different overall state. So the returned value doesn't really make a difference - I guess. But I agree with @vpereira - 100 makes sense *in case the function is called*.

If you check a2b7c8392a79c63cc88d3dc09d2e2786eb5ae16c when testing_percentage was introduced, the 0 was about the situation when there were no openqa jobs. And *that* situation meant openQA did not yet start to test. But checks that are not there, should be missing - to be relevant.

I hope that clarifies a little.